mlflow/models/evaluation/base.py

```diff
@@ -1359,6 +1366,8 @@ def evaluate(
           metrics.
         - **col_mapping**: A dictionary mapping column names in the input dataset or output
           predictions to column names used when invoking the evaluation functions.
+        - **k**: The number of top retrieved documents to use when computing the built-in metric
+          precision_at_k.
```

Comment:
not sure if this has already been discussed, but i wonder if it makes more sense for the `precision_at_k` metric to take in `k` as a parameter rather than passing it in via evaluator config. generally i feel that the evaluator config is more confusing, and that if `k` is only useful for the `precision_at_k` metric it should be configurable at that point.

Reply:
Do you mean `mlflow.metrics.precision_at_k(k=3)`? I think that is a great idea. We did not do this just because there was no way to do it.

Reply:
yea exactly. that way we dont have to pass `k` via evaluator config.

Reply:
One question for this is how to specify k with the default evaluator.

Reply:
Discussed offline, let's still pass k via evaluator config.

Reply:
We should also make it a parameter of the built in metric function

Reply:
That also makes sense. Let me try to update.